src/foremast/awslambda/awslambda.py

```diff
@@ -60,7 +60,9 @@ def __init__(self, app, env, region, prop_path):
         self.role = app.get('lambda_role') or generated.iam()['lambda_role']
         self.timeout = app['lambda_timeout']
 
-        self.role_arn = get_role_arn(self.role, self.env, self.region)
+        self.role_arn = get_role_arn(
+            self.properties[env]['app'].get('instance_profile', generated.iam()['role']),
```

Comment:
It doesn't look like `self.role` gets used anywhere. Save this long line to `self.role` and use that instead.